src/finch/interface/eager.py

```diff
@@ -0,0 +1,178 @@
+from abc import ABC
+from . import lazy
+from .lazy import LazyTensor, defer
+from .fuse import compute
+from typing import Callable, Tuple
+import numpy as np
+
+
+class EagerTensor(ABC):
+    def __add__(self, other):
+        return add(self, other)
+
+    def __radd__(self, other):
+        return add(other, self)
+
+    def __sub__(self, other):
+        return subtract(self, other)
+
+    def __rsub__(self, other):
+        return subtract(other, self)
+
+    def __mul__(self, other):
+        return multiply(self, other)
+
+    def __rmul__(self, other):
+        return multiply(other, self)
+
+    def __abs__(self):
+        return abs(self)
+
+    def __pos__(self):
+        return positive(self)
+
+    def __neg__(self):
+        return negative(self)
+
+    def __array_function__(self, func, types, args, kwargs):
+        """Override NumPy functions for EagerTensor using the __array_function__ protocol."""
+        # Map NumPy functions to EagerTensor methods or module-level functions
+        function_map = {
+            np.add: add,
+            np.subtract: subtract,
+            np.multiply: multiply,
+            np.abs: abs,
+            np.positive: positive,
+            np.negative: negative,
+            np.expand_dims: expand_dims,
+            np.squeeze: squeeze,
+            np.prod: prod,
+        }
+        if func in function_map:
+            return function_map[func](*args, **kwargs)
+        return NotImplemented
+
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        """Override NumPy ufuncs for EagerTensor using the __array_ufunc__ protocol."""
+        # Only handle the default '__call__' method
+        if method != '__call__':
+            return NotImplemented
+        # Map ufuncs to EagerTensor operations
+        ufunc_map = {
+            np.add: add,
+            np.subtract: subtract,
+            np.multiply: multiply,
+            np.negative: negative,
+            np.positive: positive,
+            np.absolute: abs,
+            np.abs: abs,
+            # Add more ufuncs as needed
+        }
+        if ufunc in ufunc_map:
+            return ufunc_map[ufunc](*inputs, **kwargs)
```

Comment:
Similar to above. We'll eventually need to handle `reduce` as well as `accumulate`.

Reply:
These are handled differently, under the `__reduce__` method, right?

Reply:
`np.add.reduce` is roughly the same as `sum`, but with different defaults. `np.add.accumulate` is the cumulative sum.

We don't need to support *every single one*, but it'd be nice to unify the interface.